common/src/program_storage.rs

```diff
@@ -54,19 +54,23 @@ pub trait ProgramStorage {
     }
 
     /// Store a program to be associated with the given key `program_id` from the map.
-    fn add_program(program_id: ProgramId, program: ActiveProgram) -> Result<(), Self::Error> {
+    fn add_program(
+        program_id: ProgramId,
+        program: ActiveProgram,
+        block_number: u32,
```

Comment:
can we go with callback for querying block number instead of explicit argument?

Reply:
or open an issue for it

Reply:
done